File: InsertionSort/InsertionSortPython/InsertionSort.py
# Insertion Sort using an array of 100 integers.

# prints array


def printArray(arr):
    for i in arr:
        print(i)

# creates array that starts at 100 and descends to 1


def createArray():
    arr = []
    for i in range(100, 0, -1):
        arr.append(i)
    return arr

# insertion sort function


def insertionSort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i-1

        while j >= 0 and key < arr[j]:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key


arr = createArray()
insertionSort(arr)
printArray(arr)

File: InsertionSort/PredefinedInputCS/InsertionSort.cs
using System;

namespace InsertionSortCSharpPredefinedInput{
    class InsertionSort{

        public static void printArray(int[] array, int size){
            for(int i = 0; i < size; i++){
                Console.WriteLine(array[i]);
            }
        }

        public static int[] sort(int[] array, int size){
            int i, j, key;

            for(i = 1; i < size; i++){
                key = array[i];
                j = i - 1;

                while( j >= 0 && array[j] > key ){
                    array[j+1] = array[j];
                    j = j - 1;
                }
                array[j+1] = key;
            }

            return array;
        }
        static void Main(String[] args){
            int size = 4;
            int[] array = new int[] { 4, 3, 2, 1 };


            array = sort(array, size);
            printArray(array, size);
        }
    }
}
